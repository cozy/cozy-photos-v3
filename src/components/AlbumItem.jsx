import styles from '../styles/albumsList'

import React, { Component } from 'react'
// import classNames from 'classnames'
import { Link, withRouter } from 'react-router-dom'
import { translate } from '../lib/I18n'

import { getPhotoLink } from '../actions/photos'

export class AlbumItem extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isLoading: true,
      isImageLoading: true
    }

    this.handleImageLoaded = this.handleImageLoaded.bind(this)
    this.fetchPhoto = this.fetchPhoto.bind(this)

    if (props.album && props.album.photosIds.length) {
      this.fetchPhoto(props.album.photosIds[0])
    } else {
      this.state = {
        url: '',
        isLoading: false
      }
    }
  }

  fetchPhoto (photoId) {
    getPhotoLink(photoId)
      .then(link => this.setState({
        url: link,
        isLoading: false
      }))
  }

  handleImageLoaded () {
    this.setState({ isImageLoading: false })
  }

  render () {
    const { t, album, router } = this.props
    const { isLoading, url, isImageLoading } = this.state
    const parentPath = router.location.pathname
    return (
      !isLoading &&
        <div className={styles['pho-album']}>
          <Link
            to={`${parentPath}/${album._id}`}
            className={styles['pho-album-link']}
          >
            <img
              className={styles['pho-album-photo-item']}
              onLoad={this.handleImageLoaded}
              style={isImageLoading ? 'display:none' : ''}
              alt={`${album.name} album cover`}
              src={url}
            />
            <h2 className={styles['pho-album-title']}>{album.name}</h2>
            <h4 className={styles['pho-album-description']}>
              {t('Albums.album_item_description',
                {smart_count: album.photosIds.length})
              }
            </h4>
          </Link>
        </div>
    )
  }
}

export default translate()(withRouter(AlbumItem))
