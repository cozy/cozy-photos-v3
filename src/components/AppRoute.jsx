import React from 'react'
import { Route, Redirect } from 'react-router'

import Layout from './Layout'
import Timeline from '../containers/Timeline'
import AlbumsView from '../containers/AlbumsView'
import AlbumPhotos from '../containers/AlbumPhotos'
import Viewer from '../components/Viewer'
import { PhotosPicker } from '../ducks/albums'

export const ComingSoon = () => (<p style='margin-left: 2em'>Coming soon!</p>)

const AppRoute = (
  <Route component={Layout}>
    <Route path='photos' component={Timeline}>
      <Route path=':photoId' component={Viewer} />
    </Route>
    <Route path='albums' component={AlbumsView}>
      <Route path='new' component={PhotosPicker} />
      <Route path=':albumId' component={AlbumPhotos}>
        <Route path='edit' component={PhotosPicker} />
        <Route path=':photoId' component={Viewer} />
      </Route>
    </Route>
    <Redirect from='/*' to='photos' />
  </Route>
)

export default AppRoute
