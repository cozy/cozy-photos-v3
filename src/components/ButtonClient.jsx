import styles from '../styles/buttonclient'

import { translate } from '../lib/I18n'

import React from 'react'

const ButtonClient = ({ t }) => (
  <a href={t('Nav.link-client')} target='_blank' className={styles['coz-btn-client']}><span>{t('Nav.btn-client')}</span></a>
)

export default translate()(ButtonClient)
