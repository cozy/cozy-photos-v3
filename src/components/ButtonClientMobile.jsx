/* global localStorage */

import styles from '../styles/buttonclient'

import { translate } from '../lib/I18n'

import React, { Component } from 'react'

import classNames from 'classnames'

class ButtonClientMobile extends Component {
  constructor (props) {
    super(props)
    this.state = {
      seen: JSON.parse(localStorage.getItem('app_ad')) || false
    }
  }

  read () {
    localStorage.setItem('app_ad', JSON.stringify(true))
    this.setState({seen: true})
  }

  render ({ t }, { seen }) {
    if (seen) return null

    return (
      <div className={styles['coz-banner-client']}>
        <a href={t('Nav.link-client')} target='_blank' className={styles['coz-btn-clientMobile']} onClick={e => this.read(e)} ><span>{t('Nav.btn-client-mobile')}</span></a>
        <button className={classNames('coz-btn', styles['coz-btn--close'])} onClick={e => this.read()} />
      </div>
    )
  }
}

export default translate()(ButtonClientMobile)
