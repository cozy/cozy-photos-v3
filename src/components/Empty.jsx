import styles from '../styles/emptyAndError'

import React from 'react'
import { translate } from '../lib/I18n'

export const Empty = ({ t, emptyType }) => {
  return (
    <div className={styles['pho-empty']}>
      <h2>{ t(`Empty.${emptyType}_title`) }</h2>
      <p>{ t(`Empty.${emptyType}_text`)}</p>
    </div>
  )
}

const TranslatedEmpty = translate()(Empty)

export const withEmpty = (isEmpty, type, BaseComponent) => props =>
  isEmpty(props)
    ? <TranslatedEmpty emptyType={type} />
    : <BaseComponent {...props} />

export default TranslatedEmpty
