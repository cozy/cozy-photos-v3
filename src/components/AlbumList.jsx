// import styles from '../styles/albumList'

import React from 'react'
import { translate } from '../lib/I18n'

import Empty from '../components/Empty'
import Topbar from '../components/Topbar'

export const PhotosList = ({ t }) => {
  return (
    <div>
      <Topbar viewName='albums' />
      <div role='contentinfo'>
        <Empty emptyType='albums' />
      </div>
    </div>
  )
}

export default translate()(PhotosList)
