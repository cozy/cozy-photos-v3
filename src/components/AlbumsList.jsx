import styles from '../styles/albumsList'

import React from 'react'

import { withEmpty } from '../components/Empty'
import AlbumItem from '../components/AlbumItem'
import NewAlbum from '../components/NewAlbum'

const DumbAlbumsList = props => (
  <div className={styles['pho-album-list']}>
    {props.albums.map((a) => <AlbumItem album={a} key={a._id} onServerError={props.onServerError} />)}
    <NewAlbum />
  </div>
)

const AlbumsList = withEmpty(props => props.albums.length === 0, 'albums', DumbAlbumsList)

export default AlbumsList
