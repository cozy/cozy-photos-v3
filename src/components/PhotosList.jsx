import styles from '../styles/photoList'

import React from 'react'
import { translate } from '../lib/I18n'

import Empty from './Empty'
import Loading from './Loading'
import SelectionBar from '../containers/SelectionBar'
import Photo from './Photo'

export const PhotosList = props => {
  const { f, photosByMonth, showSelection, selected, onPhotoToggle } = props
  const { isIndexing, isFetching, isWorking, isFirstFetch } = props
  if (isIndexing) {
    return <Loading loadingType='photos_indexing' />
  }
  if (isFetching || isFirstFetch) {
    return <Loading loadingType='photos_fetching' />
  }
  if (isWorking) {
    return <Loading loadingType='photos_upload' />
  }
  return (
    <div
      role='contentinfo'
      className={showSelection ? styles['pho-list-selection'] : ''}
    >
      {showSelection && <SelectionBar />}
      {Object.keys(photosByMonth).map(month => {
        return (
          <div className={styles['pho-section']} key={month}>
            <h3>{f(month, 'MMMM YYYY')}</h3>
            {photosByMonth[month].map(photo =>
              <Photo
                photo={photo}
                key={photo._id}
                selected={selected.indexOf(photo._id) !== -1}
                onToggle={onPhotoToggle}
              />
            )}
          </div>
        )
      })}
      {Object.keys(photosByMonth).length === 0 && <Empty emptyType='photos' />}
    </div>
  )
}

export default translate()(PhotosList)
