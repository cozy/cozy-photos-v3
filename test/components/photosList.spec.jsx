'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { mockT, mockF } from '../lib/I18n'
import { PhotosList } from '../../src/components/PhotosList'

const mockFetchedPhotos = {
  '2017-01-01': [
    {
      _id: '33dda00f0eec15bc3b3c59a615001ac8',
      created_at: '0001-01-01T00:00:00Z',
      name: 'MonImage.jpg',
      size: '150000',
      updated_at: '0001-01-01T00:00:00Z'
    }
  ]
}

describe('PhotosList component', () => {
  it('should render correctly a timeline of photos according a photos array', () => {
    const component = shallow(
      <PhotosList t={mockT} f={mockF} photosByMonth={mockFetchedPhotos} selected={[]} />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a loading view if isIndexing is true', () => {
    const component = shallow(
      <PhotosList t={mockT} f={mockF} photosByMonth={mockFetchedPhotos} selected={[]} isIndexing />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a loading view if isFetching is true', () => {
    const component = shallow(
      <PhotosList t={mockT} f={mockF} photosByMonth={mockFetchedPhotos} selected={[]} isFetching />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a loading view if isWorking is true', () => {
    const component = shallow(
      <PhotosList t={mockT} f={mockF} photosByMonth={mockFetchedPhotos} selected={[]} isWorking />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a loading view if isFirstFetch is true', () => {
    const component = shallow(
      <PhotosList t={mockT} f={mockF} photosByMonth={mockFetchedPhotos} selected={[]} isFirstFetch />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should render correctly an empty view if photos is empty', () => {
    const component = shallow(
      <PhotosList t={mockT} f={mockF} photosByMonth={[]} selected={[]} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
