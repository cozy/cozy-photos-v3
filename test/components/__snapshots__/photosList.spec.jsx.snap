exports[`PhotosList component should render correctly a loading view if isFetching is true 1`] = `
<_translate
  loadingType="photos_fetching" />
`;

exports[`PhotosList component should render correctly a loading view if isFirstFetch is true 1`] = `
<_translate
  loadingType="photos_fetching" />
`;

exports[`PhotosList component should render correctly a loading view if isIndexing is true 1`] = `
<_translate
  loadingType="photos_indexing" />
`;

exports[`PhotosList component should render correctly a loading view if isWorking is true 1`] = `
<_translate
  loadingType="photos_upload" />
`;

exports[`PhotosList component should render correctly a timeline of photos according a photos array 1`] = `
<div
  className=""
  role="contentinfo">
  <div
    className="pho-section">
    <h3>
      January 2017
    </h3>
    <withRouter(Component)
      onToggle={undefined}
      photo={
        Object {
          "_id": "33dda00f0eec15bc3b3c59a615001ac8",
          "created_at": "0001-01-01T00:00:00Z",
          "name": "MonImage.jpg",
          "size": "150000",
          "updated_at": "0001-01-01T00:00:00Z",
        }
      }
      selected={false} />
  </div>
</div>
`;

exports[`PhotosList component should render correctly an empty view if photos is empty 1`] = `
<div
  className=""
  role="contentinfo">
  <_translate
    emptyType="photos" />
</div>
`;
